name: Test
on:
  workflow_call:
    inputs:
      continue:
        description: "Continue the workflow when errors are encountered"
        required: false
        default: false
        type: boolean
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v4
      - name: Cache node modules
        id: cache
        uses: actions/cache@v4
        with:
          path: |
            node_modules
            .yarn
          key: ${{ runner.os }}-node-${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            ${{ runner.os }}-node-
      - name: Set up Nodejs
        uses: actions/setup-node@v4
        with:
          node-version: "22.16.0"
      - name: Install dependencies
        if: steps.cache.outputs.cache-hit != 'true'
        run: yarn install
      - name: Run tests
        continue-on-error: ${{ inputs.continue }}
        id: run-tests
        run: yarn test
